src/main/scala/firrtl/transforms/formal/ConvertAsserts.scala

```diff
@@ -0,0 +1,36 @@
+// See LICENSE for license details.
+
+package firrtl.transforms.formal
+
+import firrtl._
+import firrtl.ir._
+import firrtl.options.Dependency
+
+/**
+  * Convert Asserts
+  *
+  * Replaces all Assert nodes with a gated print-and-stop. This effectively
+  * emulates the assert for IEEE 1364 Verilog.
+  */
+object ConvertAsserts extends Transform with DependencyAPIMigration {
+  override def prerequisites = Nil
+  override def optionalPrerequisites = Nil
+  override def optionalPrerequisiteOf = Seq(
+    Dependency[VerilogEmitter],
+    Dependency[MinimumVerilogEmitter],
+    Dependency[RemoveVerificationStatements])
+
+  override def invalidates(a: Transform): Boolean = false
+
+  def convertAsserts(stmt: Statement): Statement = stmt match {
+    case Verification(Formal.Assert, i, clk, pred, en, msg) =>
+      val nPred = DoPrim(PrimOps.Not, Seq(pred), Nil, pred.tpe)
+      val gatedNPred = DoPrim(PrimOps.And, Seq(nPred, en), Nil, pred.tpe)
+      Block(Print(i, msg, Nil, clk, gatedNPred), Stop(i, 1, clk, gatedNPred))
```

Comment:
Should the `Print` be omitted when the assert message is empty?

Reply:
The message is mandatory, but I could certainly add a special case for the empty string. That seems like a sensible thing to add, so I'll put that in unless anyone objects.

Reply:
Added

Reply:
Thanks! This way we can fall back to emitting a printf and an assert with an empty message in the Chisel frontend in order to implement a backwards compatible assert(x, "%d", y)